pkg/virtctl/imageupload/imageupload.go

```diff
@@ -210,8 +210,7 @@ func (c *command) run(cmd *cobra.Command, args []string) error {
 	if err := parseArgs(args); err != nil {
 		return err
 	}
-	// #nosec No risk for path injection. This is used only to upload an image not to read info
-	file, err := os.Open(imagePath)
+	file, err := os.Open(filepath.Clean(imagePath))
```

Comment:
It is not clear to me what `filepath.Clean` helps. It would not make anything saver (see https://play.golang.org/p/M2cf1P__hAy) and anyway people can provide to this command any path they want. If they have permission to open it, then everything is good. virtctl does not have any additional permissions comparted to the calling user.

Even if the path would have to be validated, `filepath.Clean` alone does not make anything saver. Therefore it is not clear to my why this is suggested as a solution by the gosec manual.

Reply:
Right. I was too fast to consume the gosec example.

I've posted https://github.com/securego/securego.github.io/pull/22 per your suggestion.

Reply:
@rmohr filepath.Clean() alone indeed doesn't help, In addition to "cleaning" the path you need to make sure the resulting path is absolute path (for example using filepath.IsAbs() )

Reply:
A nice example (in addition to Roman's general one is:  `filepath.Clean("foo/../../../../../../etc/passwd"`  Which was a know web-server attack :-)

Reply:
> A nice example (in addition to Roman's general one is: `filepath.Clean("foo/../../../../../../etc/passwd"` Which was a know web-server attack :-)

You're most welcome to suggest it as a follow-up to my https://github.com/securego/securego.github.io/pull/22 (which was merged in super speed).

My main pain point here is not `filepath.Clean`, but the comment that it attempted to replace. That comment is wrong. Uploading `/etc/passwd` is bad even though you only read from the file. The comment should explain why it is ok for the `run` function to read from arbitrary files. The reason is that `run` has zero elevated permission relative to the caller of `virtctl`. Can you think of a way to express this in proper code?

Reply:
@dankenigsberg You basically say - the caller can access whatever path he can open, so you are not looking for some validation or rules and therefor there is nothing we can do in code - we can only explain it through comment. 

BTW, (not related to this case) - Maybe we can disallow using relative paths across KubeVirt (and if someone must have it - that should be the exception)  ?  Relative path is a security nightmare :-)